docs/developers_guide/framework.rst

```diff
@@ -117,80 +117,70 @@ and debugging.
+
+The ``CompassConfigParser`` class also includes methods for adding a user
+config file and other config files by file name, but these are largely intended
+for use by the framework rather than individual test cases.
 
-When this is called within the ``mesh`` test case, nothing will happen because
-``compass/ocean/tests/global_ocean/mesh`` does not contain a ``mesh.cfg`` file.
-The config files for meshes are handled differently, since they aren't
-associated with a particular test case:
+Other methods for the ``CompassConfigParser`` are similar to those for
+:py:class:`configparser.ConfigParser`.  In addition to ``get()``,
+``getinteger()``, ``getfloat()`` and ``getboolean()`` methods, this class
+implements :py:meth:`compass.config.CompassConfigParser.getlist()`, which
+can be used to parse a config value separated by spaces and/or commas into
+a list of strings, floats, integers, booleans, etc.
+
+Currently, ``CompassConfigParser`` supports accessing a config section using
+section names as keys, e.g.:
 
 .. code-block:: python
 
-    mesh_step = mesh.mesh_step
-    add_config(config, mesh_step.package, mesh_step.mesh_config_filename,
-               exception=True)
-
-In this case, the mesh step keeps track of the package and config file in its
-attributes (e.g. ``compass.ocean.tests.global_ocean.mesh.qu240`` and
-``qu240.cfg`` for the ``QU240`` and ``QUwISC240`` meshes).  Since we require
-each mesh to have config options (to define the vertical grid and the metadata
-to be added to the mesh, at the very least), we use ``exception=True`` so an
-exception will be raised if no config file is found.
-
-The ``config`` module also contains 3 functions that are intended for internal
-use by the framework itself. Test-case developers will typically not need to
-call these functions directly.
-
-The :py:func:`compass.config.duplicate_config()` function can be used to make a
-deep copy of a ``config`` object so changes can be made without affecting the
-original.
-
-The :py:func:`compass.config.ensure_absolute_paths()` function is used
-internally by the framework to check and update config options in the
-``paths``, ``namelists``, ``streams``, and ``executables`` sections of the
-config file to make sure they have absolute paths. The absolute paths are
-determined from the location where one of the tools from the compass
-:ref:`dev_command_line` was called.
-
-The :py:func:`compass.config.get_source_file()` function is used to get an
-absolute path for a file using one of the config options defined in the
-``paths`` section.  This function is used by the framework as part of
-downloading files (e.g. to a defined database), see :ref:`dev_io`.
+    section = self.config['enthalpy_benchmark_viz']
+    display_image = section.getboolean('display_image')
+    ...
+
+But it does not allow assignment of a section or many of the other
+dictionary-like features supported by :py:class:`configparser.ConfigParser`.
```

Comment:
Yes, that's exactly right.  I only implemented the `__getitem__` method and not the other dictionary methods.  They would potentially be complicated to implement because of the comment and provenance handling, and I don't foresee a need.